File: 83_RemoveDuplicatesfromSortedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null) {
			return head;
		}
        ListNode res = head;
        while(head.next != null) {
        	if(head.val == head.next.val) {
        		head.next = head.next.next;
        	}else {
				head = head.next;
			}
        }
        
        return res;
    }
}

File: c++/83_RemoveDuplicatesfromSortedList.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head) return head;
        auto cur = head;
        for (auto p = head->next; p; p = p->next) {
            if (p->val != cur->val) {
                cur = cur->next = p;
            }
        }
        cur->next = nullptr;
        return head;
    }
};
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head) return head;
        auto dummy = new ListNode(-1);
        dummy->next = head;
        auto p = dummy->next;
        while (p->next) {
            auto q = p->next;
            while (q && q->val == p->val) q = q->next;
            if (p->next == q) p = p->next;
            else p->next = q;
        }
        return dummy->next;
    }
};